SharedPackages/BrowserServicesKit/Sources/PixelKit/WidePixel/WidePixel.swift

```diff
@@ -0,0 +1,247 @@
+//
+//  WidePixel.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import os.log
+import Common
+
+#if os(iOS)
+import UIKit
+#endif
+
+public protocol WidePixelManaging {
+    func startFlow<T: WidePixelData>(_ data: T)
+    func updateFlow<T: WidePixelData>(_ data: T)
+    func completeFlow<T: WidePixelData>(_ data: T, status: WidePixelStatus, onComplete: @escaping PixelKit.CompletionBlock)
+}
+
+public final class WidePixel: WidePixelManaging {
+
+    public struct MeasuredInterval: Codable {
+        public var start: Date?
+        public var end: Date?
+
+        public init(start: Date? = nil, end: Date? = nil) {
+            self.start = start
+            self.end = end
+        }
+
+        public static func startingNow() -> MeasuredInterval {
+            return MeasuredInterval(start: Date())
+        }
+
+        public mutating func complete(at date: Date = Date()) {
+            self.end = date
+        }
+    }
+
+    private static let logger = Logger(subsystem: "PixelKit", category: "Wide Pixel")
+    private static let storageQueue = DispatchQueue(label: "com.duckduckgo.wide-pixel.storage-queue", qos: .utility)
+
+    private let storage: WidePixelStoring
+    private let pixelKitProvider: () -> PixelKit?
+    private let sampler: WidePixelSampling
+    private let eventMapping: EventMapping<WidePixelEvent>?
+
+    public init(storage: WidePixelStoring = WidePixelUserDefaultsStorage(),
+                pixelKitProvider: @escaping () -> PixelKit? = { PixelKit.shared },
+                sampler: WidePixelSampling? = nil,
+                events: EventMapping<WidePixelEvent>? = nil) {
```

Comment:
The event mapper is emitting events, but not actually being used yet. I'll write it up in the PRs that add the subscription pixel integration.